### ISSUE TITLE
Publish TypeScript definition files to enable TypeScript support

### ISSUE THREAD

**Issue by dpoetzschke** [talon-one/talon_one.js#17]
Hello,

as of TypeScript 3.7 it is possible to create TypeScript definition files from JavaScript sources as long as JSDoc is provided.
See https://www.typescriptlang.org/docs/handbook/declaration-files/dts-from-js.html

To enable TypeScript support for Talon_one.js it would be beneficial if we could have the definition files without creating and maintaining them ourself outside of this repository.

The changes needed are provided within the link but it should be something like this:
`package.json`
```
{
  ...
  "typings": "types",
  "scripts": {
    "prepack": "babel src -d dist && typescript src/**/*.js --declaration --allowJs --emitDeclarationOnly --outDir types",
    ...
  },
  "devDependencies": {
    "typescript": "^3.9.7"
    ....
  }
}
```

Thanks in advance

**Comment by altJake** [comment]
Hi @dpoetzschke,

Firstly, I would like to apologize that it took me so long to answer.
And secondly, I would like to thank you for the great suggestion and for the help with providing the simple straightforward change in the publishing process.

I would like to add you suggestion as part of the next release to the library, and added the suggestion in the RC-branch ([v4.2.0](https://github.com/talon-one/talon_one.js/tree/v4.2.0)).
I have tested it myself and validated that it does provide _some_ definitions and autocomplete suggestions in Visual Studio Code, but maybe you could also try it and confirm that it would provide the result that you were wishing for? (cc/ @NazarYablonskiy)

Installing it directly from that branch could be done with the following command/s (_please notice that `git` must present on the machine that the command is run from_):
```shell
# npm
npm install talon-one/talon_one.js#v4.2.0

# yarn
yarn add talon-one/talon_one.js#v4.2.0
```

P.S.
@dpoetzschke and @NazarYablonskiy,
Considering that you both were asking about Typescript support, maybe we could work together in order to provide a better TS support for our API?

Currently, in order to automate our SDKS generation process, we are using [OpenAPI Generator](https://openapi-generator.tech/) to generate our different SDK, including this JS package.

The idea is to use one of the supported [typescript client generators](https://github.com/OpenAPITools/openapi-generator/tree/master/samples/client/petstore) by the tool, and generate a full-featured Typescript package (that would probably replace this current one).
In order to get to a library that is both easy to use and integrate with different setups, I would like to maybe consult with you around the different options and together come to a consensus which one will serve those.
Would you have some time to checkout different samples and conduct some (written or live) feedback sessions to achieve that?

A polite no will be more than understandable and acceptable answer :)

**Comment by dpoetzschke** [comment]
Hello @altJake 

thanks for the fast integration.

As for your suggestion:
I am not familiar with the openAPI Generator. But if you already use the JS generator shouldn't it be only the validation that all types are then given?
I am not sure how much of a help i could be considering we are only going to use certain parts of the API (of which im also not that familiar with)

**Comment by altJake** [comment]
Hi @dpoetzschke ,

It's totally fine that you are not familiar with the OpenAPI Generator, you help could be in the shape of helping me figuring out which implementation of it is the most straightforward to integrated in a typescript-based backend system.

We are also using typescript, but not in our backend, so as much as I have some experience with integration TS/JS libraries into a codebase - I believe that your input/feedback will be extremely useful.

I have already played around with the different generator "flavors", and my idea was to share a few code snippets of how the code on your end would look like using any of those. Then we will probably find the best one to allow you most comfort using it and us while releasing fully-supported TS versions of the SDK.

So your help would be basically choosing your future API of our TS-SDK , and again, as much as I will appreciate the help, it will be totally understandable if you don't have the time for it at the moment

**Comment by dpoetzschke** [comment]
Sure thing i can do this.
We are running very strict rules for typescript with full types etc. So i can surely tell you how each of the solutions would feel in our environment.

**Comment by altJake** [comment]
Amazing, very much appreciated 🙏 

I will close this issue now that #20 is merged and [v4.2.0](https://github.com/talon-one/talon_one.js/releases/tag/v4.2.0) is released.

@dpoetzschke once I get to set up the groundwork for the different flavors and examples of typescript code I will ping you again.

Thanks a lot in advance!
